metrics/perplexity/perplexity.py

```diff
@@ -0,0 +1,149 @@
+# coding=utf-8
+# Copyright 2021 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Perplexity Metric """
+
+import torch
+from datasets.utils import tqdm
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+import datasets
+
+
+_CITATION = """\
+
+"""
+
+_DESCRIPTION = """
+Perplexity (PPL) is one of the most common metrics for evaluating language models.
+It is defined as the exponentiated average negative log-likelihood of a sequence.
+
+For more information, see https://huggingface.co/docs/transformers/perplexity
+"""
+
+_KWARGS_DESCRIPTION = """
+Args:
+    model_id (str): model used for calculating Perplexity
+            NOTE: Perplexity can only be calculated for causal language models.
+                    This includes models such as gpt2, causal variations of bert,
+                    causal versions of t5, and more (the full list can be found
+                    in the AutoModelForCausalLM documentation here:
+                    https://huggingface.co/docs/transformers/master/en/model_doc/auto#transformers.AutoModelForCausalLM )
+
+    input_texts (list of str): input text, each separate text snippet
+        is one list entry. Perplexity returned will be an average of
+        the perplexity for each list entry.
+    stride (int): stride size, defaults to 512
+    device (str): device to run on, defaults to 'cuda' when available
+Returns:
+    perplexity: dictionary containing the average perplexity score for the text
+        in the input list.
+Examples:
+    Example 1:
+        >>> perplexity = datasets.load_metric("perplexity")
+        >>> input_texts = ["lorem ipsum", "Happy Birthday!", "Bienvenue"]
+        >>> results = perplexity.compute(model_id='gpt2',
+        ...                              input_texts=input_texts,
+        ...                              stride=1)
+        >>> round(results["perplexity"], 2)
+        837.04
+
+    Example 2:
+        >>> perplexity = datasets.load_metric("perplexity")
+        >>> input_texts = datasets.load_dataset("wikitext", "wikitext-2-raw-v1", "test") # doctest:+ELLIPSIS
+        [...]
+        >>> results = perplexity.compute(model_id='gpt2',
+        ...                              input_texts=input_texts,
+        ...                              stride=256)
+        >>> round(results["perplexity"], 2)
+        304.81
```

Comment:
```suggestion
        107.75
```

Reply:
Fixed the code so that when the data is loaded, it's loaded as:
`input_texts = datasets.load_dataset("wikitext", "wikitext-2-raw-v1", split="test")["text"]`

This should change it so that the output is 304.81 on your end, too (but if you could confirm that, that would be great!)

Reply:
Cool thanks !